### PR TITLE
Model part reordering

### PR DESCRIPTION
**Pull request by RiccardoRossi**
adding a function to reorder the modelpart.

so far this gave around 10% speedup, but it has the potential for more on badly ordered problems

### CODE REVIEW

File: kratos/processes/reorder_and_optimize_modelpart_process.cpp [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+            {
+                const auto& cond = mrModelPart.ConditionsBegin() + i;              
+                const auto& geom = cond->GetGeometry();
+                for(unsigned int k=0; k<geom.size(); ++k)
+                    for(unsigned int l=0; l<geom.size(); ++l)
+                        graph[geom[k].Id()-1].insert(geom[l].Id()-1); //the -1 is because the ids of our nodes start in 1
+            }
+            
+             
+            unsigned int nnz = 0;
```
Comment (loumalouomega) [inline kratos/processes/reorder_and_optimize_modelpart_process.cpp]: What is nnz?

**Review by loumalouomega — APPROVED** [review]
Looks OK, maybe some int can be defined like unsigned int and some variable I don't understand

File: kratos/processes/reorder_and_optimize_modelpart_process.cpp (hunk 1 above)
Comment (RiccardoRossi) [reply kratos/processes/reorder_and_optimize_modelpart_process.cpp]: nnz = Number of Non Zeros. I think it is commonly called nnz in this context (at least within ublas)

Comment (RiccardoRossi) [reply kratos/processes/reorder_and_optimize_modelpart_process.cpp]: just to tell that i added a comment explaining that

**Pull request merged**